dask/dataframe/reshape.py

```diff
@@ -356,13 +357,16 @@ def melt(
 
     from dask.dataframe.core import no_default
 
-    return frame.map_partitions(
-        M.melt,
-        meta=no_default,
-        id_vars=id_vars,
-        value_vars=value_vars,
-        var_name=var_name,
-        value_name=value_name,
-        col_level=col_level,
-        token="melt",
-    )
+    # during melt, we might combine different datatypes into the same
+    # object column. We shouldn't force those objects to string[pyarrow]
+    with config.set({"dataframe.convert_string": False}):
```

Comment:
With `convert_string` on, `melt` would produce a weird and unexpected result:

```python
import pandas as pd
import dask.dataframe as dd
from dask import config

config.set({"dataframe.convert_string": True})
df = pd.DataFrame({"x": list("abc") * 2, "y": list("xy") * 3, "c": [2, 3, 1, 5, 6, 4]})
ddf = dd.from_pandas(df, 2)

res_pd = pd.melt(df)
res_dd = dd.melt(ddf).compute()
print(f"{res_pd.value.tolist() = }")
print(f"{res_dd.value.tolist() = }")
```

Produces:

```
res_pd.value.tolist() = ['a', 'b', 'c', 'a', 'b', 'c', 'x', 'y', 'x', 'y', 'x', 'y', 2, 3, 1, 5, 6, 4]
res_dd.value.tolist() = ['a', 'b', 'c', 'x', 'y', 'x', '2', '3', '1', 'a', 'b', 'c', 'y', 'x', 'y', '5', '6', '4']
```

Because `pandas` used `object` dtype, integers were preserved. With.`dask`, we would force them into `string[pyarrow]`. What would make sense is upcasting them to the common dtype, which is `object`.

To me, suppressing pyarrow strings during `melt` seemed like a safer choice. I don't know how many people actually use `melt` though. Perhaps if we don't have `pyarrow` strings as default, we can just xfail the test for now and get back to it later.

Reply:
I see, thanks for the extra context. For now I'd prefer to `xfail` the failing tests instead of disabling. We may ultimately end up disabling as is currently done here. But I'd rather consider behavior here along with the other follow-up items